eventheader_macros/src/provider_generator.rs

```diff
@@ -128,16 +128,13 @@ impl ProviderGenerator {
             // #[no_mangle]
             .add_punct("#")
             .add_group_square(self.tree1.add_ident("no_mangle").drain())
-            // static mut _eh_define_provider_MY_PROVIDER: *const usize = ::core::ptr::null();
+            // static _eh_define_provider_MY_PROVIDER: *const usize = ::core::ptr::null();
             .add_ident("static")
-            .add_ident("mut")
```

Comment:
I believe this is the only line that matters in this PR. I think you could revert all of the others and the fix would still work.

IIRC, this variable is never actually dereferenced. It exists only for error-checking -- it gives you a compile error if you misspell your provider name, and it gives you a link error if two crates have conflicting provider names. There is a lot of flexibility in what we do with it. However, it is currently stored in a linker section that stores MUT variables, which is why I marked it MUT. That probably doesn't matter -- removing the MUT probably doesn't do anything. In theory the right thing to do would be to make it a Cell or something. But this is fine.